pkgs/development/cuda-modules/_cuda/db/schema.nix

```diff
@@ -0,0 +1,608 @@
+          type = attrsOf (submodule {
+            options = {
+              clang = mkOption {
+                type = VersionRange;
+                description = "Range of supported Clang compilers";
+              };
+              gcc = mkOption {
+                type = VersionRange;
+                description = "Range of supported GCC compilers";
+              };
+            };
+          });
+          description = ''
+            `∷ ProductVersion ⇒ { gcc : MajorVersionRange, clang: MajorVersionRange }`
+
+            Host compilers expected by NVCC in each CUDAToolkit release.
+          '';
+          example = lib.listToAttrs (lib.take 1 (lib.attrsToList bootstrapData.nvccCompatibilities));
+        };
+
+      release = {
+        product = mkOption {
+          type = SetOfStr;
+          default = { };
+          example = {
+            "cuda" = 1;
+            "cudnn" = 1;
+            "tensorrt" = 1;
+          };
+          description = ''
+            `∷ ProductName ⇒ ()`
+
+            Corresponds to `release_product` in NVIDIA manifests
+          '';
+        };
+
+        package = mkColumnOption cudb.product (attrsOf (attrsOf (nullOr str))) {
+          default = { };
+          description = ''
+            `∷ ProductName ⇒ ProductVersion ⇒ PName ⇒ Maybe PackageVersion`
+
+            Packages (`pname`, `version`) published in each release (`(productName, productVersion)`),
+            or, in NVIDIA manifests, `(release_product, release_label)`)
+          '';
+        };
+      };
+
+      archive.sha256 = mkOption {
+        description = "`∷ SHA256 ⇒ { URL, SystemStringNvidia, CompatibilityTag ⇒ () }`";
+        default = { };
+        type = attrsOf (submodule {
+          options = {
+            systemNv = mkOption {
+              type = enum (builtins.attrNames cudb.system.nvidia);
+              description = "`∷ SystemStringNvidia`";
+            };
+            extraConstraints = mkOption {
+              default = { };
+              description = "`∷ CompatibilityTag ⇒ ()`";
+              type = attrsOf (
+                submodule (
+                  { name, ... }:
+                  {
+                    options = {
+                      "<" = lib.mkOption {
+                        type = nullOr str;
+                        default = null;
+                        description = ''
+                          `∷ VersionString`
+
+                          Package or product ${name} must be `versionOlder version ${name}.version`.
+                        '';
+                      };
+                      ">=" = lib.mkOption {
+                        type = nullOr str;
+                        default = null;
+                        description = ''
+                          `∷ VersionString`
+
+                          Package or product ${name} must be `versionAtLeast version ${name}.version`.
+                        '';
+                      };
+                    };
```

Comment:
`null` values are filtered out somewhere else with `apply`, right?

Reply:
They are not filtered out for `_cuda.db`, but they are for `_rawManifests.export`